vital/types/detected_object_set.h

```diff
@@ -220,6 +220,26 @@ class VITAL_EXPORT detected_object_set
   detected_object_set_sptr select( const std::string& class_name,
                                    double             threshold = detected_object_type::INVALID_SCORE ) const;
 
+  /**
+   * @brief Filter detections via an arbitrary function
+   *
+   * Filter the detected object set based on some predicate function
+   *
+   * @param p Predicate function which returns true if an element should
+   *          be removed.
+   *
+   * @return List of detections.
```

Comment:
It looks like this method returns VOID not a list of detections.
It may be helpful to indicate that the filter exits the current set of detections.